Adopt BEWebContentFilter API for content filtering

szewai opened the pull request:
#### e67b2b3fb4e92995fd54d173e2c3331a07477fa2
<pre>
Adopt BEWebContentFilter API for content filtering
<a href="https://bugs.webkit.org/show_bug.cgi?id=302428">https://bugs.webkit.org/show_bug.cgi?id=302428</a>
<a href="https://rdar.apple.com/163939782">rdar://163939782</a>

Reviewed by Chris Dumez.

Introduce WebParentalControlsURLFilter, which inherits ParentalControlsURLFilter and implements content filtering with
BEWebContentFilter API.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/platform/cocoa/ContentFilterUnblockHandlerCocoa.mm:
(WebCore::ContentFilterUnblockHandler::requestUnblockAsync):
* Source/WebCore/platform/cocoa/ParentalControlsContentFilter.h:
* Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm:
(WebCore::ParentalControlsContentFilter::protectedImpl const):
(WebCore::ParentalControlsContentFilter::enabled const):
(WebCore::ParentalControlsContentFilter::responseReceived):
(WebCore::ParentalControlsContentFilter::impl const): Deleted.
* Source/WebCore/platform/cocoa/ParentalControlsURLFilter.h:
* Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
(WebCore::allFiltersWithConfigurationPath):
(WebCore::ParentalControlsURLFilter::filterWithConfigurationPath):
(WebCore::ParentalControlsURLFilter::ParentalControlsURLFilter): Deleted.
(WebCore::ParentalControlsURLFilter::singleton): Deleted.
(WebCore::globalQueue): Deleted.
(WebCore::webContentFilterTypeDidChange): Deleted.
(WebCore::registerNotificationForWebContentFilterTypeChange): Deleted.
(WebCore::ParentalControlsURLFilter::resetIsEnabled): Deleted.
(WebCore::ParentalControlsURLFilter::isWCRBrowserEngineClientEnabled const): Deleted.
(WebCore::ParentalControlsURLFilter::isEnabled const): Deleted.
(WebCore::ParentalControlsURLFilter::isURLAllowed): Deleted.
(WebCore::ParentalControlsURLFilter::allowURL): Deleted.
(WebCore::ParentalControlsURLFilter::effectiveWCRBrowserEngineClient): Deleted.
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::setSharedParentalControlsURLFilterIfNecessary):
(WebKit::NetworkResourceLoader::startContentFiltering):
* Source/WebKit/Shared/ios/WebParentalControlsURLFilter.h: Added.
* Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm: Added.
(WebKit::WebParentalControlsURLFilter::create):
(WebKit::WebParentalControlsURLFilter::ensureWebContentFilter):
(WebKit::WebParentalControlsURLFilter::isEnabledImpl const):
(WebKit::WebParentalControlsURLFilter::isURLAllowed):
(WebKit::WebParentalControlsURLFilter::allowURL):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/303237@main">https://commits.webkit.org/303237@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ddb685370c60d75b5ac8f3dca8463a3738ea652

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131623 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3955 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42589 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139132 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83420 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6e7f5f14-8392-47cd-8eb0-ff082f444da2) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133493 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4165 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3796 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100488 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68066 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7e83f56b-2bd0-43df-ae80-d97d61031f50) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134569 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117814 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81296 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/cd9929e7-216e-4420-aa78-7db1978bf236) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2888 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/524 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82323 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/123648 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/111620 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35940 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141777 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/130077 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3699 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36456 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108862 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3747 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3281 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109106 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27672 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2805 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114143 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56907 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3760 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32544 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/163094 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3587 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67171 "Built successfully") | [⏳ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Tests-EWS "Waiting to run tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3842 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3690 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -51,13 +50,13 @@ static bool wcrBrowserEngineClientEnabled(const String& path)
 }
 
-ParentalControlsURLFilter& ParentalControlsURLFilter::filterWithConfigurationPath(const String& path)
+Ref<ParentalControlsURLFilter> ParentalControlsURLFilter::sharedFilterWithConfigurationPath(const String& path)
```
Why does this return a Ref now? This doesn't seem to follow our usual style.

cdumez replied:
And if you need an overload that returns a Ref, to make Safer CPP happy at some call sites, you can add one and use `protectedSharedFilterWithConfigurationPath` as name, like we usually do.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -82,13 +81,28 @@ static bool wcrBrowserEngineClientEnabled()
 }
 
-ParentalControlsURLFilter::ParentalControlsURLFilter() = default;
+Ref<ParentalControlsURLFilter> ParentalControlsURLFilter::sharedFilter()
```
Ditto, why does this return a `Ref<>` and not a `ParentalControlsURLFilter&`, like `singleton()` used to return. We're not transferring ownership here.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -82,13 +81,28 @@ static bool wcrBrowserEngineClientEnabled()
+
+void ParentalControlsURLFilter::setSharedFilter(Ref<ParentalControlsURLFilter>&& filter)
+{
+    globalFilter() = filter.copyRef();
```
`globalFilter() = WTFMove(filter);` would be more efficient.

cdumez replied:
Do we ever overwrite an existing `globalFilter` here? If not, we could add a RELEASE_ASSERT that `globalFilter()` is null before assignment and keep the `singleton()` naming so that we don't have to ref the filter everywhere.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -82,13 +81,28 @@ static bool wcrBrowserEngineClientEnabled()
+    globalFilter() = filter.copyRef();
+}
+
+ParentalControlsURLFilter::ParentalControlsURLFilter()
```
`= default;`

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm:
```diff
@@ -54,12 +54,12 @@
 #if HAVE(WEBCONTENTRESTRICTIONS)
 
-ParentalControlsURLFilter& ParentalControlsContentFilter::impl() const
+Ref<ParentalControlsURLFilter> ParentalControlsContentFilter::impl() const
```
Why does this return a `Ref` now? This doesn't seem like a step in the right direction.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -121,7 +135,11 @@ static void registerNotificationForWebContentFilterTypeChange()
 }
 
-bool ParentalControlsURLFilter::isWCRBrowserEngineClientEnabled() const
+ParentalControlsURLFilter::~ParentalControlsURLFilter()
```
`= default;`

cdumez commented on Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
```diff
@@ -284,12 +288,23 @@ void NetworkResourceLoader::startRequest(const ResourceRequest& newRequest)
+{
+#if HAVE(BROWSERENGINEKIT_WEBCONTENTFILTER)
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [] {
```
Can this get called on multiple threads? I doubt it. If not, a static bool would suffice, we don't need the locks inside `call_once`.

cdumez commented on Source/WebKit/Shared/ios/WebParentalControlsURLFilter.h:
```diff
@@ -0,0 +1,54 @@
+
+private:
+    WebParentalControlsURLFilter();
+    ~WebParentalControlsURLFilter();
```
No strong reason to make this destructor private I think.

cdumez commented on Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm:
```diff
@@ -0,0 +1,104 @@
+    return adoptRef(*new WebParentalControlsURLFilter);
+}
+
+WebParentalControlsURLFilter::WebParentalControlsURLFilter()
```
`= default;`

cdumez commented on Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm:
```diff
@@ -0,0 +1,104 @@
+{
+}
+
+WebParentalControlsURLFilter::~WebParentalControlsURLFilter()
```
`= default;`

cdumez commented on Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm:
```diff
@@ -0,0 +1,104 @@
+        if (!m_webContentFilter)
+            lazyInitialize(m_webContentFilter, adoptNS([[BEWebContentFilter alloc] init]));
+
+        [m_webContentFilter evaluateURL:url.createNSURL().get() completionHandler:makeBlockPtr([url, weakFilter = WTFMove(weakFilter)](BOOL shouldBlock, NSData *replacementData) mutable {
```
Why are we capturing the `url` in the block here? I don't see it being used. Also, if we really need to capture it, are we sure the block gets called on the exact same queue? If not, we likely need to isolate copy.

cdumez commented on Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm:
```diff
@@ -0,0 +1,104 @@
+    workQueue().dispatch([this, protectedThis = Ref { *this }, currentIsEnabled = isEnabled(), url = crossThreadCopy(url), completionHandler = WTFMove(completionHandler)]() mutable {
+        assertIsCurrent(workQueue());
+        if (!currentIsEnabled) {
+            ensureOnMainRunLoop([completionHandler = WTFMove(completionHandler)] mutable {
```
`callOnMainRunLoop` is better here, you already know you can't be on the main thread I think.

cdumez commented on Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm:
```diff
@@ -0,0 +1,104 @@
+        if (!m_webContentFilter)
+            lazyInitialize(m_webContentFilter, adoptNS([[BEWebContentFilter alloc] init]));
+
+        [m_webContentFilter allowURL:url.createNSURL().get() completionHandler:makeBlockPtr([url, completionHandler = WTFMove(completionHandler)](BOOL didAllow, NSError *) mutable {
```
Why do we need to capture the `url` in the block here?

cdumez commented on Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm:
```diff
@@ -0,0 +1,104 @@
+
+        [m_webContentFilter allowURL:url.createNSURL().get() completionHandler:makeBlockPtr([url, completionHandler = WTFMove(completionHandler)](BOOL didAllow, NSError *) mutable {
+            RELEASE_LOG(Loading, "WebParentalControlsURLFilter::allowURL result %d.\n", didAllow);
+            ensureOnMainRunLoop([didAllow, completionHandler = WTFMove(completionHandler)] mutable {
```
Same comment about using `callOnMainThread` if you already know you can't be on the main thread.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm:
```diff
@@ -68,7 +68,7 @@
 {
 #if HAVE(WEBCONTENTRESTRICTIONS)
-    return impl().isEnabled();
+    return Ref { impl() }->isEnabled();
```
Seems like we should add a `protectedImpl()` getter and call it here...

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm:
```diff
@@ -110,7 +110,7 @@ static inline bool canHandleResponse(const ResourceResponse& response)
     m_evaluatedURL = response.url();
     m_state = State::Filtering;
-    impl().isURLAllowed(*m_evaluatedURL, *this);
+    Ref { impl() }->isURLAllowed(*m_evaluatedURL, *this);
```
... and here.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -82,12 +81,26 @@ static bool wcrBrowserEngineClientEnabled()
+static RefPtr<ParentalControlsURLFilter>& globalFilter()
 {
-    static MainThreadNeverDestroyed<UniqueRef<ParentalControlsURLFilter>> filter = UniqueRef(*new ParentalControlsURLFilter);
+    static NeverDestroyed<RefPtr<ParentalControlsURLFilter>> filter;
```
Why did you get rid of the `MainThreadNeverDestroyed` ? Is this now accessed / modified from multiple threads? If so, it seems like we would need a Lock?

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -104,7 +117,7 @@ static void webContentFilterTypeDidChange(CFNotificationCenterRef, void*, CFStri
         filter->resetIsEnabled();
 #else
-    ParentalControlsURLFilter::singleton().resetIsEnabled();
+    Ref { ParentalControlsURLFilter::singleton() }->resetIsEnabled();
```
We should not need the `Ref { }` here due to the `singleton()` naming. Static analysis should be happy without it.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -184,11 +199,11 @@ static void registerNotificationForWebContentFilterTypeChange()
 {
 #if HAVE(WEBCONTENTRESTRICTIONS_PATH_SPI)
-    auto& filter = WebCore::ParentalControlsURLFilter::filterWithConfigurationPath(parameters.configurationPath);
+    Ref filter { WebCore::ParentalControlsURLFilter::filterWithConfigurationPath(parameters.configurationPath) };
```
I think the `=` initialization is more common in WebKit for local variables.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -184,11 +199,11 @@ static void registerNotificationForWebContentFilterTypeChange()
+    Ref filter { WebCore::ParentalControlsURLFilter::filterWithConfigurationPath(parameters.configurationPath) };
 #else
-    auto& filter = WebCore::ParentalControlsURLFilter::singleton();
+    Ref filter { WebCore::ParentalControlsURLFilter::singleton() };
```
Ditto.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
```diff
@@ -207,6 +222,11 @@ static void registerNotificationForWebContentFilterTypeChange()
 
+WorkQueue& ParentalControlsURLFilter::workQueue()
+{
+    return globalQueue();
```
We should rename `globalQueue()` to `globalQueueSingleton()` so static analysis can know it is a singleton.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.h:
```diff
@@ -38,27 +40,32 @@ namespace WebCore {
 #endif
     WEBCORE_EXPORT static void allowURL(const ParentalControlsURLFilterParameters&, CompletionHandler<void(bool)>&&);
-
+    WEBCORE_EXPORT WorkQueue& workQueue();
```
Seems like this could be static and renamed to `workQueueSingleton()` no? This way static analysis would know it doesn't need ref'ing.

cdumez commented on Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
```diff
@@ -284,12 +288,24 @@ void NetworkResourceLoader::startRequest(const ResourceRequest& newRequest)
+static void setSharedParentalControlsURLFilterIfNecessary()
+{
+#if HAVE(BROWSERENGINEKIT_WEBCONTENTFILTER)
+    static bool initialized = false;
```
Can we please add an assertion to make sure this is only called on the main thread?

cdumez commented on Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm:
```diff
@@ -0,0 +1,100 @@
+            return;
+        }
+
+        if (!m_webContentFilter)
```
Maybe we should add an `ensureWebContentFilter()` member function since this lazy initialization logic is duplicated in 2 places.

cdumez commented on Source/WebCore/platform/cocoa/ParentalControlsURLFilter.h:
```diff
@@ -38,27 +40,32 @@ namespace WebCore {
 #endif
     WEBCORE_EXPORT static void allowURL(const ParentalControlsURLFilterParameters&, CompletionHandler<void(bool)>&&);
-
+    WEBCORE_EXPORT static WorkQueue& workQueueSignleton();
```
Typo: workQueueSignleton

cdumez reviewed (approved):
LGTM with typo fix.

szewai committed:
Adopt BEWebContentFilter API for content filtering
https://bugs.webkit.org/show_bug.cgi?id=302428
rdar://163939782

Reviewed by Chris Dumez.

Introduce WebParentalControlsURLFilter, which inherits ParentalControlsURLFilter and implements content filtering with
BEWebContentFilter API.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/platform/cocoa/ContentFilterUnblockHandlerCocoa.mm:
(WebCore::ContentFilterUnblockHandler::requestUnblockAsync):
* Source/WebCore/platform/cocoa/ParentalControlsContentFilter.h:
* Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm:
(WebCore::ParentalControlsContentFilter::protectedImpl const):
(WebCore::ParentalControlsContentFilter::enabled const):
(WebCore::ParentalControlsContentFilter::responseReceived):
(WebCore::ParentalControlsContentFilter::impl const): Deleted.
* Source/WebCore/platform/cocoa/ParentalControlsURLFilter.h:
* Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
(WebCore::allFiltersWithConfigurationPath):
(WebCore::ParentalControlsURLFilter::filterWithConfigurationPath):
(WebCore::ParentalControlsURLFilter::ParentalControlsURLFilter): Deleted.
(WebCore::ParentalControlsURLFilter::singleton): Deleted.
(WebCore::globalQueue): Deleted.
(WebCore::webContentFilterTypeDidChange): Deleted.
(WebCore::registerNotificationForWebContentFilterTypeChange): Deleted.
(WebCore::ParentalControlsURLFilter::resetIsEnabled): Deleted.
(WebCore::ParentalControlsURLFilter::isWCRBrowserEngineClientEnabled const): Deleted.
(WebCore::ParentalControlsURLFilter::isEnabled const): Deleted.
(WebCore::ParentalControlsURLFilter::isURLAllowed): Deleted.
(WebCore::ParentalControlsURLFilter::allowURL): Deleted.
(WebCore::ParentalControlsURLFilter::effectiveWCRBrowserEngineClient): Deleted.
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::setSharedParentalControlsURLFilterIfNecessary):
(WebKit::NetworkResourceLoader::startContentFiltering):
* Source/WebKit/Shared/ios/WebParentalControlsURLFilter.h: Added.
* Source/WebKit/Shared/ios/WebParentalControlsURLFilter.mm: Added.
(WebKit::WebParentalControlsURLFilter::create):
(WebKit::WebParentalControlsURLFilter::ensureWebContentFilter):
(WebKit::WebParentalControlsURLFilter::isEnabledImpl const):
(WebKit::WebParentalControlsURLFilter::isURLAllowed):
(WebKit::WebParentalControlsURLFilter::allowURL):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/303237@main
